Flaky test_cancel_job for process pool CFEngine

`test_cancel_job` in `test_cf_engine.py` has flaked twice in the nightlies recently.

https://github.com/alteryx/evalml/runs/3179257203?check_suite_focus=true

https://github.com/alteryx/evalml/runs/3212514044?check_suite_focus=true

This is the stack trace:

```
    @pytest.mark.parametrize("pool_type", ["threads", "processes"])
    def test_cancel_job(X_y_binary_cls, pool_type, thread_pool, process_pool):
        """Test that training a single pipeline using the parallel engine produces the
        same results as simply running the train_pipeline function."""
        X, y = X_y_binary_cls
        pool = get_pool(pool_type, thread_pool, process_pool)
    
        with CFClient(pool) as client:
            engine = CFEngine(client=client)
            pipeline = DaskPipelineSlow({"Logistic Regression Classifier": {"n_jobs": 1}})
    
            # Verify that engine fits a pipeline
            pipeline_future = engine.submit_training_job(
                X=X, y=y, automl_config=automl_data, pipeline=pipeline
            )
            pipeline_future.cancel()
>           assert pipeline_future.is_cancelled
E           assert False
E            +  where False = <evalml.automl.engine.cf_engine.CFComputation object at 0x7fa66c368610>.is_cancelled
```

I saw this failing on conda forge, so I went ahead and fixed it.  ADditionally, this failure is tied to the processes pool executor, which I think is always troublesome, even with Dask.

```
=================================== FAILURES ===================================
__________________________ test_cancel_job[processes] __________________________
[gw0] linux -- Python 3.7.11 /home/runner/work/evalml/evalml/test_python/bin/python

X_y_binary_cls = (          0         1         2   ...        17        18        19
0  -0.039268  0.131912 -0.211206  ...  1.976989  ...ns], 0     0
1     0
2     1
3     1
4     1
     ..
95    1
96    1
97    1
98    1
99    0
Length: 100, dtype: int64)
pool_type = 'processes'
```

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -6,6 +6,7 @@ Release Notes
     * Changes
     * Documentation Changes
     * Testing Changes
+        * Removes the process-level parallelism from the ``test_cancel_job`` test :pr:`2666`
 
 .. warning::
 

--- a/evalml/tests/automl_tests/parallel_tests/test_cf_engine.py
+++ b/evalml/tests/automl_tests/parallel_tests/test_cf_engine.py
@@ -342,7 +342,7 @@ def test_submit_scoring_jobs_multiple(
         np.testing.assert_allclose(par_scores, seq_scores, rtol=1e-10)
 
 
-@pytest.mark.parametrize("pool_type", ["threads", "processes"])
+@pytest.mark.parametrize("pool_type", ["threads"])
 def test_cancel_job(X_y_binary_cls, pool_type, thread_pool, process_pool):
     """Test that training a single pipeline using the parallel engine produces the
     same results as simply running the train_pipeline function."""

